### PR TITLE
feat: add closeChannels API to server wallet

### PR DESCRIPTION
**Pull request by snario**

### CODE REVIEW

File: packages/server-wallet/src/wallet/index.ts [hunk 1]
```diff
@@ -421,6 +421,40 @@ export class Wallet extends EventEmitter<WalletEvent>
...
+        data: {targetChannelId: channelId, fundingStrategy: channel.fundingStrategy},
+      };
+
+      const objectiveToStore: ObjectiveStoredInDB = {
+        ...objective,
+        status: 'approved',
+        objectiveId: objectiveId(objective),
+      };
+
+      await ObjectiveModel.insert(objectiveToStore, this.knex);
```
Comment (tomclose) [inline packages/server-wallet/src/wallet/index.ts:444]: Should this use `this.store.addObjective`?

Comment (snario) [reply packages/server-wallet/src/wallet/index.ts:444]: Yes

File: packages/server-wallet/src/wallet/index.ts [hunk 2]
```diff
@@ -391,6 +391,44 @@ export class Wallet extends EventEmitter<WalletEvent>
     return this.store.lockApp(channelId, criticalCode, handleMissingChannel);
   }
 
+  async closeChannels(channelIds: Bytes32[]): Promise<MultipleChannelOutput> {
```
Comment (kerzhner) [inline packages/server-wallet/src/wallet/index.ts:394]: Should this leverage the `closeChannel` api defined below?

File: packages/server-wallet/src/wallet/index.ts [hunk 3]
```diff
@@ -391,6 +391,44 @@ export class Wallet extends EventEmitter<WalletEvent>
...
+            CloseChannel.CloseChannelError.reasons.channelMissing,
+            {channelId}
+          );
+        }
+      );
+    }
+
+    const {channelResults, outbox} = await this.takeActions(channelIds);
+
+    (outbox[0].params.data as Payload).objectives = channelIds.map(channelId => ({
```
Comment (kerzhner) [inline packages/server-wallet/src/wallet/index.ts:423]: Is it possible the type assertion?

Comment (snario) [reply packages/server-wallet/src/wallet/index.ts:446]: The `Store` creates an outbound objective right now and the type of this is `unknown` which is strategically set (cc @tomclose) for outbound objects. One solution might be to put this line of code inside the store... but that would create an even bigger mess and contribute to the code smell of the store handling communications. I think this type assertion is OK for now.

File: packages/server-wallet/src/wallet/__test__/integration/close-channel.test.ts [hunk 4]
```diff
@@ -57,3 +58,28 @@ it("reject when it's not my turn", async () => {
...
+      channelNonce: i,
+      vars: [stateWithHashSignedBy([alice(), bob()])(runningState)],
+    });
+    await Channel.query(w.knex).insert(c);
+    channelIds.push(c.channelId);
+  }
+
+  await expect(w.closeChannels(channelIds)).resolves.toMatchObject({
+    channelResults: channelIds.map(_ => ({
+      status: 'closing',
```
Comment (geoknee) [inline packages/server-wallet/src/wallet/__test__/integration/close-channel.test.ts:80]: Can you put the `channelId` in each object of this array?

File: packages/server-wallet/src/wallet/index.ts (hunk 3 above)
Comment (geoknee) [reply packages/server-wallet/src/wallet/index.ts:453]: Seems to me that the wallet _should_ be using the `Payload` type. It should be `unknown` only from the point of view of the app / messaging layer (outside the wallet's interface).

**Review by geoknee — APPROVED** [review]

File: packages/server-wallet/src/wallet/index.ts (hunk 3 above)
Comment (snario) [reply packages/server-wallet/src/wallet/index.ts:453]: There should probably be a cleanup of the wallet code and `takeActions` to account for this, maybe a new issue

**Pull request merged**